crates/holochain_state/src/source_chain.rs

```diff
@@ -626,113 +626,117 @@ where
+                let agent_pubkey = self.agent_pubkey().clone();
+                move |txn| -> Result<_, DatabaseError> {
+                    // prepare sql statement depending on provided cap secret
+                    let mut statement = match &check_secret {
+                        Some(cap_secret) => {
+                            // cap grant for cap secret must exist
+                            // that has not been updated or deleted
+                            let sql = "
+                                SELECT Entry.blob
+                                FROM Entry
+                                WHERE Entry.cap_secret = ?1
+                                AND (
+                                    SELECT COUNT(Action.hash)
+                                    FROM Action
+                                    WHERE Action.author = ?2
+                                    AND (
+                                        Action.original_entry_hash = Entry.hash
+                                        OR Action.deletes_entry_hash = Entry.hash
+                                    )
+                                ) = 0
+                            "
+                            .trim();
+                            let mut statement = txn.prepare(sql)?;
+                            let cap_secret_blob = to_blob(cap_secret).map_err(|err| {
+                                DatabaseError::SerializedBytes(SerializedBytesError::Serialize(
+                                    err.to_string(),
+                                ))
+                            })?;
+                            statement.raw_bind_parameter(1, cap_secret_blob)?;
+                            statement.raw_bind_parameter(2, agent_pubkey.clone())?;
+                            statement
+                        }
+                        None => {
+                            // unrestricted cap grant must exist
+                            // that has not been updated or deleted
+                            let sql = "
+                                SELECT blob
+                                FROM Entry
+                                WHERE access_type = ?1
+                                AND (
+                                    SELECT COUNT(Action.hash)
+                                    FROM Action
+                                    WHERE Action.author = ?2
+                                    AND (
+                                        Action.original_entry_hash = Entry.hash
+                                        OR Action.deletes_entry_hash = Entry.hash
+                                    )
+                                ) = 0;
+                            "
+                            .trim();
+                            let mut statement = txn.prepare(sql)?;
+                            statement.raw_bind_parameter(1, "unrestricted")?;
+                            statement.raw_bind_parameter(2, agent_pubkey.clone())?;
```

Comment:
We don't use `raw_bind_parameter` anywhere else, I wonder why you're using it here?

In any case I think it would be better to use the `CapAccess::Unrestricted` type directly instead of the `"unrestricted"` string, since that will catch the case that we ever rename or remove the variant.

Reply:
Consider instead, something like:

```rust
let res = if let Some(cap_secret) = &check_secret {
    txn.query_row("MY SQL 1", params![cap_secret, yada])
} else {
    txn.query_row("MY SQL 2", params![CapAccess::Unrestricted, yada])
};
```

Reply:
alas, looks like we're using hard-coded strings for the enums: https://github.com/holochain/holochain/blob/develop/crates/holochain_state/src/mutations.rs#L661

If you want to fix this, you could implement those to/from sql on that enum... just make sure rusqlite isn't imported into the integrity types crate.

Reply:
> If you want to fix this, you could implement those to/from sql on that enum... just make sure rusqlite isn't imported into the integrity types crate.

I did that and also replaced the hardcoded string in `mutations.rs` by a string representation of the variant identifier. Let me know if that makes sense or not.